File: go-code/extract_variable/extract-variable.go
package extract_variable

import (
	"gitlab.com/clean-code55/refactor/improving-design-of-existing-code/go-code/models"
	"math"
)

func price(order models.Order) float64 {
	// price = base price - quantity discount + shipping
	return float64(order.Quantity)*order.ItemPrice -
		math.Max(0, float64(order.Quantity-500))*0.05 +
		math.Min(float64(order.Quantity)*order.ItemPrice*0.1, 100)
}

File: java-code/src/main/java/refactoring/extract_variable/ExtractVariable.java
package refactoring.extract_variable;

import refactoring.model.Order;

public class ExtractVariable {

  public double price(Order order) {
    //price = base price - quantity discount + shipping
    return order.getQuantity() * order.getItemPrice() -
        Math.max(0, order.getQuantity() - 500) * 0.05 +
        Math.min(order.getQuantity() * order.getItemPrice()* 0.1, 100);
  }
}
